kubernetes/annotations_and_labels.md

```diff
@@ -8,6 +8,9 @@ This page defines common annotations and labels we set in Kubernetes objects.
   location. For now this should be in our `giantswarm/giantswarm` repository in
   which we crosslink all pages to create a reasonable documentation of all kinds
   of Kubernetes resources, their functionality and relationships.
+- `giantswarm.io/restart` - this annotation is meant to be used for DaemonSet,
+  Deployment and StatefulSet restarts. The value can be arbitrary string that
```

Comment:
Maybe I am a dummy here but I don't get the use case from this description - when would this annotation be added?

Reply:
+1

Reply:
> when would this annotation be added?

When you want to programmatically issue a rolling restart of `Deployment`, `DaemonSet` or `StatefulSet`. I.e. when you want to implement `kubectl rollout restart` in some operator.

In my use-case I want to perform this in `azure-operator` during worker VMSS upgrade in order to gracefully move workload from old nodes to new nodes.

Reply:
Let's write that this should be added manually by the user in order to perform rolling restart.